### PR TITLE
SX Design: add new component `Menu`

### PR DESCRIPTION
**Pull request by mrtnzlml**
Please note: this component is not finished yet, but it's a good start.
I need to focus on correct positioning and accessibility in the future
commits.

Inspiration:

- https://seek-oss.github.io/braid-design-system/components/MenuItem
- https://vercel.com/design/menu

---

Screen Shot 2021-09-27 at 10 10 34

### CODE REVIEW

File: src/sx-design/src/Menu/Menu.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * TODO: render in a `Drawer` for small viewports (see `ModalDrawer`)
+ * TODO: use positioning algorithm similar to `Tooltip` (reuse)
+ * TODO: accessibility (https://www.w3.org/TR/wai-aria-practices-1.1/#menu)
+ */
+export default function Menu(props: Props): Node {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const handleButtonClick = useCallback(() => {
```
Comment (tbergq) [inline src/sx-design/src/Menu/Menu.js:30]: Why are you using `useCallback` here?

File: src/sx-design/src/Menu/__tests__/Menu.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      </Menu.Item>
+      <Menu.ItemDivider data-testid="MenuItemDivider" />
+      <Menu.Item tint="error" onClick={jest.fn()}>
+        Delete
+      </Menu.Item>
+    </Menu>,
+  );
+
+  // Everything is hidden by default (not in the DOM):
+  expect(queryByText('One')).toBeNull();
```
Comment (tbergq) [inline src/sx-design/src/Menu/__tests__/Menu.test.js:27]: What about using [toBeInTheDocument](https://testing-library.com/docs/guide-disappearance/#nottobeinthedocument)

File: src/sx-design/src/Menu/__tests__/Menu.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  expect(queryByText('Two')).toBeNull();
+  expect(queryByText('Three')).toBeNull();
+  expect(queryByTestId('MenuItemDivider')).toBeNull();
+  expect(queryByText('Delete')).toBeNull();
+
+  // We have to first open the menu:
+  fireEvent.click(getByRole('button'));
```
Comment (tbergq) [inline src/sx-design/src/Menu/__tests__/Menu.test.js:34]: Probably not the most important, but there is a package that is better for user events https://testing-library.com/docs/ecosystem-user-event/

File: src/sx-design/src/Menu/Menu.js (hunk 1 above)
Comment (mrtnzlml) [reply src/sx-design/src/Menu/Menu.js:30]: "Just in case we are going to optimize `Button` one day" kind of thinking.

But you are right it's probably not a good idea to do it for no reason. Will remove (as well as from the future PRs where I did something simialr).

**Review by tbergq — APPROVED** [review]

File: src/sx-design/src/Menu/__tests__/Menu.test.js (hunk 3 above)
Comment (mrtnzlml) [reply src/sx-design/src/Menu/__tests__/Menu.test.js:34]: I will have a look at it thanks! Since I am not familiar with this package: do you know how does it differ from the `fireEvent`? In other words: why does `fireEvent` even exist if it's better to use something else? 🤔

File: src/sx-design/src/Menu/__tests__/Menu.test.js (hunk 2 above)
Comment (mrtnzlml) [reply src/sx-design/src/Menu/__tests__/Menu.test.js:27]: I was not aware of this function, thanks!

This brings a question though. I was always using something like this in my tests:

```js
expect(getByText('Click me!')).toBeDefined();
```

Are you saying it's better to use this?

```js
expect(getByText('Click me!')).toBeInTheDocument();
```

And of course, if yes, do you know why? What is the difference? 🤔 

Thanks!

File: src/sx-design/src/Menu/__tests__/Menu.test.js (hunk 3 above)
Comment (tbergq) [reply src/sx-design/src/Menu/__tests__/Menu.test.js:34]: I think userEvent also triggers hover, mouseOver etc before clicking the button, while the fireEvent does not. I think they were planning on merging this into the main package at some point. 

I'll see if I can find the article for you once I have time.

Comment (mrtnzlml) [reply src/sx-design/src/Menu/__tests__/Menu.test.js:34]: I am already preparing a PR to change the whole package from `fireEvent` to `userEvent`! 👍

File: src/sx-design/src/Menu/__tests__/Menu.test.js (hunk 2 above)
Comment (tbergq) [reply src/sx-design/src/Menu/__tests__/Menu.test.js]: I didn't dive into the specifics, but at least from my pov it reads better. 

There might be some other benefit/edge case it covers 🤷‍♂️
